src/lib/Response/NotificationResponse.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Response;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Custom response type for notifications to facilitate type checking when needed.
+ */
+class NotificationResponse extends Response
+{
+    /**
+     * Successful response.
+     *
+     * @param string $content
+     *
+     * @return static
+     */
+    public static function success(string $content)
+    {
+        return new static($content);
+    }
+
+    /**
+     * Error response.
+     *
+     * @param string $content
+     *
+     * @return static
+     */
+    public static function error(string $content)
+    {
+        return new static($content, static::HTTP_BAD_REQUEST);
```

Comment:
I think it's a bad idea to _restrict_ the HTTP status code here. Because of that we would loose information on why the request failed and those information might be critical for the frontend app to correctly handle some situations.